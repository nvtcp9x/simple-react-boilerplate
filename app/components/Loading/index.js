import * as React from 'react';
import styled from 'styled-components'

const StyledDiv = styled.div`
  width: 100%;
  height: 100%;
  padding-bottom: 100%;
  position: relative;
  background: #f6f8fa;
`

const StyledIframe = styled.iframe`
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  position: absolute;
  display: block;
  margin: auto;
`

const Loading = () => (
  <React.Fragment>
    <StyledDiv>
      <StyledIframe src="https://giphy.com/embed/jAYUbVXgESSti" frameBorder="0" className="giphy-embed" allowFullScreen />
    </StyledDiv>
  </React.Fragment>
)

export default Loading
