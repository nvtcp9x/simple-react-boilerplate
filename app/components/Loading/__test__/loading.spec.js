import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import Loading from '../index';

Enzyme.configure({ adapter: new Adapter() });

describe('EmptyList component', () => {
  const component = shallow(<Loading />);

  it('should ...', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
