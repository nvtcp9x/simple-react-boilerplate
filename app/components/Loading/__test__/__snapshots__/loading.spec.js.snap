// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyList component should ... 1`] = `
<React.Fragment>
  <styled.div>
    <styled.iframe
      allowFullScreen={true}
      className="giphy-embed"
      frameBorder="0"
      src="https://giphy.com/embed/jAYUbVXgESSti"
    />
  </styled.div>
</React.Fragment>
`;
